src/test/java/cat/udl/eps/softarch/academicrecruit/steps/UpdateCandidateStepDefs.java

```diff
@@ -0,0 +1,57 @@
+package cat.udl.eps.softarch.academicrecruit.steps;
+
+import cat.udl.eps.softarch.academicrecruit.domain.Candidate;
+import cat.udl.eps.softarch.academicrecruit.domain.Participant;
+import cat.udl.eps.softarch.academicrecruit.repository.CandidateRepository;
+import cat.udl.eps.softarch.academicrecruit.repository.ParticipantRepository;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.When;
+import org.json.JSONObject;
+import org.springframework.http.MediaType;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.patch;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.hamcrest.Matchers.is;
+
+public class UpdateCandidateStepDefs {
+
+    final StepDefs stepDefs;
+    final CandidateRepository candidateRepository;
+    private String newResourceUri;
+
+    public UpdateCandidateStepDefs(StepDefs stepDefs, CandidateRepository candidateRepository) {
+        this.stepDefs = stepDefs;
+        this.candidateRepository = candidateRepository;
+    }
+
+
+    @When("I change the name of the candidate to {string}")
+    public void iChangeNameOfCandidateTo(String name) throws Throwable {
+
+        Candidate candidate = new Candidate();
+        candidate.setName(name);
+        candidateRepository.save(candidate);
+
+        newResourceUri = "/candidates/"+ candidate.getId();
+
+        stepDefs.result = stepDefs.mockMvc.perform(
+                patch(newResourceUri)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(stepDefs.mapper.writeValueAsString(candidate))
+                        .accept(MediaType.APPLICATION_JSON)
+                        .with(AuthenticationStepDefs.authenticate())
+        ).andDo(print());
+    }
+
+    @And("The previously updated candidate has now name {string}")
+    public void thePreviouslyUpdatedCandidateHasNowName(String name) throws Throwable {
```

Comment:
You can check, reusing existing step, that there is no candidate with the old name and that there is a candidate with the new one. In both cases, you can use a findByName, no need to go through the API in the @Given and @Then steps